codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientHandshaker.java

```diff
@@ -227,24 +229,30 @@ public WebSocketClientHandshaker setForceCloseTimeoutMillis(long forceCloseTimeo
         return this;
     }
 
+    /**
+     * Returns a {@link ChannelPromise} of handshake process.
+     */
+    public ChannelPromise handshakePromise() {
```

Comment:
This should return a `ChannelFuture` as otherwise someone could complete it from the outside.